formulaic/transforms/basis_spline.py

```diff
@@ -136,12 +150,21 @@ def basis_spline(  # pylint: disable=dangerous-default-value  # always replaced
                 raise ValueError(
                     f"Invalid value for `df`. `df` must be greater than {degree + (1 if include_intercept else 0)} [`degree` (+ 1 if `include_intercept` is `True`)]."
                 )
-            knots = list(
-                numpy.quantile(x, numpy.linspace(0, 1, nknots + 2))[1:-1].ravel()
+            if knots_x.shape[0] == 0:
```

Comment:
After adding a test where the selection above removed all datapoints, so that `numpy.nanquantile([], [0, 1])` was raising an opaque error, I changed this condition to require at least 1 data point.

Reply:
Looks good. Thanks @bashtage!